mllib/src/main/scala/org/apache/spark/mllib/fpm/Prefixspan.scala

```diff
@@ -0,0 +1,183 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.mllib.fpm
+
+import org.apache.spark.rdd.RDD
+
+/**
+ *
+ * A parallel PrefixSpan algorithm to mine sequential pattern.
+ * The PrefixSpan algorithm is described in
+ * [[http://web.engr.illinois.edu/~hanj/pdf/span01.pdf]].
```

Comment:
Document that this implementation only covers itemsets of size 1 (Han et al define an itemset to be a subset of all items and a sequential pattern to be an ordered sequence of itemsets).

Also document whether this implementation supports non-contiguous (subsequences which skip items from the original sequence between each item) sequential patterns since Han et al define sequential patterns to be non-contiguous subsequences (Section 2: Def of **subsequence**) and if I recall correctly from our phone conversation this implementation will only support contiguous subsequences.